src/main/java/org/terracotta/statistics/extended/StatisticRegistry.java

```diff
@@ -0,0 +1,230 @@
+/*
+ * Copyright Terracotta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terracotta.statistics.extended;
+
+import org.terracotta.context.ContextManager;
+import org.terracotta.context.TreeNode;
+import org.terracotta.context.extended.OperationStatisticDescriptor;
+import org.terracotta.context.extended.ValueStatisticDescriptor;
+import org.terracotta.context.query.Matcher;
+import org.terracotta.context.query.Matchers;
+import org.terracotta.statistics.OperationStatistic;
+import org.terracotta.statistics.Table;
+import org.terracotta.statistics.Time;
+import org.terracotta.statistics.ValueStatistic;
+import org.terracotta.statistics.archive.Sample;
+
+import java.io.Serializable;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.EnumSet;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import java.util.function.Supplier;
+
+import static java.util.stream.Collectors.toList;
+import static org.terracotta.context.query.Matchers.attributes;
+import static org.terracotta.context.query.Matchers.context;
+import static org.terracotta.context.query.Matchers.hasAttribute;
+import static org.terracotta.context.query.Matchers.identifier;
+import static org.terracotta.context.query.Matchers.subclassOf;
+import static org.terracotta.context.query.QueryBuilder.queryBuilder;
+import static org.terracotta.statistics.SuppliedValueStatistic.supply;
+
+/**
+ * This class replaces the previous {@link StatisticRegistry}
+ * in the cases where you do not need any sampling and history.
+ * <p>
+ * This class typically does a sort of mapping between the registrations and the discovered
+ * operations or passthrough statistics.
+ * <p>
+ * This class also supporte the generation of management metadata from the discovered statistics.
+ * <p>
+ * Non thread-safe.
+ *
+ * @author Mathieu Carbou
+ */
+public class StatisticRegistry {
+
+  protected static final Object ME = new Object();
```

Comment:
Protected?

Reply:
so that sub-classes like in tc-platform can also provide a:

```
public static StatisticRegistry noop() {
    return new StatisticRegistry(ME, Time::absoluteTime);
  }
```

Reply:
ok